### ISSUE TITLE
reference implementation fails to reject pulses with a proportion of zero

### ISSUE THREAD

**Issue by grahamgower** [popsim-consortium/demes-spec#166]
We have no test cases with a pulse proportion of zero, and the reference implementation erroneously accepts such pulse proportions. Reported by @molpopgen over at https://github.com/popsim-consortium/demes-python/issues/576

**Comment by grahamgower** [comment]
It looks like this issue also affects ancestry proportions. I.e. the following is accepted but should not be.

```yaml
time_units: generations
demes:
- name: a
  epochs:
  - {start_size: 100, end_time: 50}
- name: b
  epochs:
  - {start_size: 100, end_time: 50}
- name: c
  ancestors: [a, b]
  proportions: [0.0, 1.0]
  start_time: 100
  epochs:
  - {start_size: 100}

```

**Comment by grahamgower** [comment]
Fixed by #167.
